tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -47,12 +48,50 @@ struct DiagRestore {
   ~DiagRestore() { Ctx.setDiagnosticHandler(OrigHandler, OrigDiagContext); }
 };
 
+static void HashAndUpdate(DxilContainerHeader *Container) {
+  // Compute hash and update stored hash.
+  // Hash the container from this offset to the end.
+  static const uint32_t DXBCHashStartOffset =
+      offsetof(struct DxilContainerHeader, Version);
+  const unsigned char *DataToHash =
+      (const unsigned char *)Container + DXBCHashStartOffset;
+  unsigned AmountToHash = Container->ContainerSizeInBytes - DXBCHashStartOffset;
+  ComputeHashRetail(DataToHash, AmountToHash, Container->Hash.Digest);
+}
+
+static void HashAndUpdateOrCopy(uint32_t Flags, IDxcBlob *Shader,
+                                IDxcBlob **Hashed) {
+  if (Flags & DxcValidatorFlags_InPlaceEdit) {
+    HashAndUpdate((DxilContainerHeader *)Shader->GetBufferPointer());
+    *Hashed = Shader;
+    Shader->AddRef();
+  } else {
+    // Possible gotcha: the blob allocated here is tied to this .dll, so the
+    // DLL shouldn't be unloaded before the blob is released.
```

Comment:
That's generally how returned blobs work in the DXC APIs.  Do we need to point this out as a comment here?  It probably belongs in the API documentation, and potentially as a comment on the API definition.

Reply:
Removed.